data/libs/functions/functions.lua

```diff
@@ -909,3 +909,47 @@ function Player:doCheckBossRoom(bossName, fromPos, toPos)
 	end
 	return true
 end
+
+function CheckDustLevel(monsterForge, player)
```

Comment:
I think that we can create a new method to Monster instead using this as global function.

I will work on this and i will give you some examples